pkg/cache/azure_cache.go

```diff
@@ -62,28 +62,56 @@ func cacheKeyFunc(obj interface{}) (string, error) {
 	return obj.(*AzureCacheEntry).Key, nil
 }
 
+// Resource operations
+type Resource interface {
+	Get(key string, crt AzureCacheReadType) (interface{}, error)
+	GetWithDeepCopy(key string, crt AzureCacheReadType) (interface{}, error)
+	Delete(key string) error
+	Set(key string, data interface{})
+	Update(key string, data interface{})
+
+	GetStore() cache.Store
+	Lock()
+	Unlock()
+}
+
 // TimedCache is a cache with TTL.
 type TimedCache struct {
-	Store  cache.Store
-	Lock   sync.Mutex
+	Store     cache.Store
+	MutuxLock sync.Mutex
+	TTL       time.Duration
+
+	ApiCaller Resource
```

Comment:
can we rename apicaller to resourceProvider? because we can also assign mock impl to it.

Reply:
done